pkg/reconciler/pipelinerun/resources/pipelineref_test.go

```diff
@@ -291,26 +291,46 @@ func TestGetPipelineFunc_RemoteResolution(t *testing.T) {
 	ctx := context.Background()
 	cfg := config.FromContextOrDefaults(ctx)
 	ctx = config.ToContext(ctx, cfg)
-	pipeline := parse.MustParseV1beta1Pipeline(t, pipelineYAMLString)
 	pipelineRef := &v1beta1.PipelineRef{ResolverRef: v1beta1.ResolverRef{Resolver: "git"}}
 
 	testcases := []struct {
 		name         string
 		pipelineYAML string
+		wantPipeline *v1beta1.Pipeline
+		wantErr      bool
 	}{{
 		name: "v1beta1 pipeline",
 		pipelineYAML: strings.Join([]string{
 			"kind: Pipeline",
 			"apiVersion: tekton.dev/v1beta1",
 			pipelineYAMLString,
 		}, "\n"),
+		wantPipeline: parse.MustParseV1beta1Pipeline(t, pipelineYAMLString),
+	}, {
+		name: "v1beta1 pipeline with beta features",
+		pipelineYAML: strings.Join([]string{
+			"kind: Pipeline",
+			"apiVersion: tekton.dev/v1beta1",
+			pipelineYAMLStringWithBetaFeatures,
+		}, "\n"),
+		wantPipeline: parse.MustParseV1beta1Pipeline(t, pipelineYAMLStringWithBetaFeatures),
 	}, {
 		name: "v1 pipeline",
 		pipelineYAML: strings.Join([]string{
 			"kind: Pipeline",
 			"apiVersion: tekton.dev/v1",
 			pipelineYAMLString,
 		}, "\n"),
+		wantPipeline: parse.MustParseV1beta1Pipeline(t, pipelineYAMLString),
+	}, {
+		name: "v1 pipeline with beta features",
+		pipelineYAML: strings.Join([]string{
+			"kind: Pipeline",
+			"apiVersion: tekton.dev/v1",
+			pipelineYAMLStringWithBetaFeatures,
+		}, "\n"),
+		wantPipeline: nil,
+		wantErr:      true,
```

Comment:
I would prefer to separate the error tests and non-error cases in 2 different test functions. like TestGetPipelineFunc_RemoteResolution_Error and TestGetPipelineFunc_RemoteResolution_Success

Reply:
thanks for the feedback-- I'm not sure if there's a strong reason to prefer one or the other so I'm going to leave as is.